src/main.js

```diff
@@ -171,4 +200,61 @@ document.addEventListener('DOMContentLoaded', function() {
       }
     }
   });
+
+  function getNotePreferences() {
+    return componentRelay.getComponentDataValueForKey('notes') || {};
+  }
+
+  function getCurrentNotePreferences() {
+    const notesPreferences = getNotePreferences();
+    return notesPreferences[lastUUID];
+  }
+
+  function setNotePreferences(key, value) {
+    const notesPreferences = getNotePreferences();
+    notesPreferences[lastUUID] = {
+      [key]: value
+    };
+    componentRelay.setComponentDataValueForKey('notes', notesPreferences);
```

Comment:
Hmm, why are we not setting this preference on the note item itself via `clientData`?

Reply:
I think it's on the contrary: if a note is duplicated, its clientData is duplicated. However if you save the preference on the component itself (as is being done now), and the note is duplicated, then this data won't carry over.

Reply:
Ah, got it. Will use `clientData` then 👍🏽